### PR TITLE
FluentCalendar, FluentDatePicker and FluentTimePicker components

### PR DESCRIPTION
**Pull request by dvoituron**
# FluentCalendar, FluentDatePicker and FluentTimePicker components

## 📖 Description

This PR adds the **FuentCalendar**, **FluentDatePicker** and **FluentTimePicker** components, as well as the associated unit tests.

The calendar features a **Culture** property to display days in the user's language. By default, the OS culture is used. It is possible to disable certain dates using the **DisabledDateFunc** function, which returns a boolean, depending on the date proposed. If required, the developer can fully customize the visual of a day via the **DaysTemplate**.

An extension method **ToTimeAgo(TimeSpan delay)** returns a string with one of these samples, depending of the delay: "Just now", "25 seconds ago", "14 minutes ago", ...

DatePicker
DatePicker
TimePicker

### General

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-blazor/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [x] I have added a new component
- [ ] I have modified an existing component

## Breaking changes

<html>
<body>
<div fluid-config-type="Tabular"><div></div>

Old FluentCalendar | New FluentCalendar
-- | --
✅ Readonly | No change - Case updated to ReadOnly to be coherent.
✅ Locale | ⚠️ Culture This property is a CultureInfo object more precise to customize rendering (including first day of week: Sunday, Monday, ...).Migration from Locale ("fr-FR") to Culture: System.Globalization.CultureInfo.GetCultureInfo("fr-FR")
✅ Month & Year | PickerMonth This new property is more Blazor/C#, because it's a DateTime object. And a two-way bindable property.
✅ DayFormat | No change
🚫 WeekdayFormat | ⚠️Breaking change - No implementedUse the Abbreviated value of .NET system weekday names.
🚫 MonthFormat | ⚠️Breaking change - No implementedUse the Full month name.
🚫 MinWeeks | ⚠️Breaking change - No implemented
🚫 OutOfMonthSelectable | ⚠️Breaking change - No implemented
✅ DisabledDates & DisabledSelectable | DisabledDateFunc This property is a function. More interesting if you have a lot of dates to deactivate. You can create a function to get these dates from a DB and not fill a large array of dates in memory.DisabledSelectable True to display the Strikethrough style. False to display the day like other, but not selectable.
⬜ SelectedDates & SelectedDatesChanged | SelectedDate & SelectedDateChangedAt this moment, only one date is selectable
✅ OnDateClicked | No change
🚫 ChildContent | ⚠️Breaking change - Removed

</div>
</body>
</html>

### CODE REVIEW

File: src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentCalendar.razor.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+namespace Microsoft.Fast.Components.FluentUI;
+
+/// <summary>
+/// Fluent Calendar based on
+/// https://github.com/microsoft/fluentui/blob/master/packages/web-components/src/calendar/.
+/// </summary>
+public partial class FluentCalendar : FluentCalendarBase
+{
+    public static string ArrowUp = "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\" fill=\"var(--neutral-fill-strong-focus)\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"M4.2 10.73a.75.75 0 001.1 1.04l5.95-6.25v14.73a.75.75 0 001.5 0V5.52l5.95 6.25a.75.75 0 001.1-1.04l-7.08-7.42a1 1 0 00-1.44 0L4.2 10.73z\"/></svg>";
```
Comment (vnbaaij) [inline src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentCalendar.razor.cs:12]: Can/should we use available icons here?

File: src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentDatePicker.razor.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.AspNetCore.Components.Web;
+using Microsoft.AspNetCore.Components;
+using System.Globalization;
+using Microsoft.Fast.Components.FluentUI.Utilities;
+
+namespace Microsoft.Fast.Components.FluentUI;
+
+public partial class FluentDatePicker : FluentCalendarBase
+{
+    public static string CalendarIcon = "<svg slot=\"end\" width=\"20\" height=\"20\" viewBox=\"0 0 24 24\" fill=\"var(--neutral-fill-strong-focus)\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"M17.75 3C19.55 3 21 4.46 21 6.25v11.5c0 1.8-1.46 3.25-3.25 3.25H6.25A3.25 3.25 0 013 17.75V6.25C3 4.45 4.46 3 6.25 3h11.5zm1.75 5.5h-15v9.25c0 .97.78 1.75 1.75 1.75h11.5c.97 0 1.75-.78 1.75-1.75V8.5zm-11.75 6a1.25 1.25 0 110 2.5 1.25 1.25 0 010-2.5zm4.25 0a1.25 1.25 0 110 2.5 1.25 1.25 0 010-2.5zm-4.25-4a1.25 1.25 0 110 2.5 1.25 1.25 0 010-2.5zm4.25 0a1.25 1.25 0 110 2.5 1.25 1.25 0 010-2.5zm4.25 0a1.25 1.25 0 110 2.5 1.25 1.25 0 010-2.5zm1.5-6H6.25c-.97 0-1.75.78-1.75 1.75V7h15v-.75c0-.97-.78-1.75-1.75-1.75z\"/>";
```
Comment (vnbaaij) [inline src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentDatePicker.razor.cs:10]: Same here, use available fluent icon?

File: src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentTimePicker.razor.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    /// <summary />
+    protected override string? StyleValue => new StyleBuilder()
+        .AddStyle(Style)
+        .Build();
+
+    /// <summary>
+    /// Gets or sets the design of this input.
+    /// </summary>
+    [Parameter]
+    public virtual TextFieldAppearance Appearance { get; set; } = TextFieldAppearance.Outline;
```
Comment (vnbaaij) [inline src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentTimePicker.razor.cs:18]: Do we want to have a separate enum for this? Values are already available in `Appearance` enum. We can check if valid value was supplied in OnParameterSet(Async). Using 'just' Appearance keeps us consistent with web-component side.

File: src/Microsoft.Fast.Components.FluentUI/Enums/TextFieldAppearance.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Microsoft.Fast.Components.FluentUI;
+
+/// <summary />
+public enum TextFieldAppearance
```
Comment (vnbaaij) [inline src/Microsoft.Fast.Components.FluentUI/Enums/TextFieldAppearance.cs:4]: See earlier remark. Do we need this?

File: src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentCalendar.razor.cs (hunk 1 above)
Comment (dvoituron) [reply src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentCalendar.razor.cs:12]: Yes. When the "Icons PR" will be merge, I will update this section.
In the meantime, I allow the developer to modify this (static) value if he wishes to change the icon.

File: src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentTimePicker.razor.cs (hunk 3 above)
Comment (dvoituron) [reply src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentTimePicker.razor.cs:18]: The **Appearance** enum contains many other values not available in the [text-field](https://github.com/microsoft/fluentui/blob/master/packages/web-components/src/text-field/text-field.vscode.definition.json) web components.

I find it confusing to allow the developer to select a value that has no result or triggers an exception. In my opinion, the other "Text Field" components should be modified to use this reduced enumeration.

Comment (vnbaaij) [reply src/Microsoft.Fast.Components.FluentUI/Components/DateTime/FluentTimePicker.razor.cs:18]: Ok, maybe I'm clinging to much to the Fluent web components implementation here and now is a good time to diverge a bit.

**Comment by LuohuaRain** [comment]
I am wondering if the value of FluentDatePicker should be of the `DateOnly` type, and if the value of FluentTimePicker should be of the `TimeOnly` type?

**Pull request merged**